fedimint-core/src/api.rs

```diff
@@ -613,6 +619,54 @@ impl WsClientConnectInfo {
     }
 }
 
+impl FromStr for WsClientConnectInfo {
+    type Err = anyhow::Error;
+
+    fn from_str(encoded: &str) -> Result<Self, Self::Err> {
+        let (hrp, data, variant) = bech32::decode(encoded)?;
+
+        ensure!(hrp == "fed", "Invalid HRP in bech32 encoding");
+        ensure!(variant == Bech32m, "Expected Bech32m encoding");
+
+        let bytes: Vec<u8> = Vec::<u8>::from_base32(&data)?;
+        let total_len = bytes.len() as u64;
+        let mut cursor = Cursor::new(bytes);
+        let mut id_bytes = [0; PK_SIZE];
+        cursor.read_exact(&mut id_bytes)?;
+
+        let mut urls = vec![];
+        while cursor.position() < total_len {
```

Comment:
https://doc.rust-lang.org/std/io/struct.Cursor.html#method.is_empty

Reply:
Needed to update rust, thought it was still unstable.

Reply:
Oh wait still is...